src/axom/primal/geometry/BoundingBox.hpp

```diff
@@ -100,6 +100,17 @@ class BoundingBox
   AXOM_HOST_DEVICE
   BoundingBox(const PointType* pts, int n);
 
+  /*!
+   * \brief Constructor. Creates a bounding box containing the
+   * initializer list of points.
+   *
+   * \param [in] pts an initializer list containing points
+   */
+  AXOM_HOST_DEVICE
+  explicit BoundingBox(std::initializer_list<PointType> pts)
```

Comment:
Design question: Will this constructor get confused with the constructor that takes a min and max point? `BoundingBox<double, 1> {Point<double, 1> {1.0}, Point<double, 1> {0.0}}`

Reply:
since you use 'explicit', I don't believe so. I may be wrong but I believe the compiler interprets and initializer_list as a single object while the ctor taking two points is considered a different overload. I recommend trying a simple example with several different compilers to be sure.

Reply:
This may provide some insight: https://en.cppreference.com/w/cpp/language/converting_constructor

Reply:
It could be a little confusing and it does lead to a subtle change in behavior.

```
BoundingBox{Point {}, Point {}}; // Calls initializer_list constructor (previously would call two point constructor)
BoundingBox(Point {}, Point {}); // Calls two point constructor
```

Reply:
Would it be better to have a make_bounding_box free function?

Reply:
I suppose the constructor from two points can be more computationally efficient when you know that one point is the min and the other is the max, and you explicitly use the third parameter (`shouldFixBounds=false`) -- but obviously this is explicit.
Otherwise, we should get the same result and I also wouldn't expect a meaningful performance difference (although, I suppose we should benchmark/profile it if it matters):
```cpp
// 1: explicit .ctor -- explicitly set min and max, but then check and fix
BoundingBoxType bbox (point1, point2, /* shouldFixBounds=*/ true);
// 2: explicit .ctor -- most efficient, but possibly unsafe
BoundingBoxType bbox (point1, point2, /* shouldFixBounds=*/ false);
// 3: initializer-list -- will ensure that bounds are correct by inserting points one at a time
BoundingBoxType bbox {point1, point2}; 
// 4: .ctor -- implicitly sets shouldFixBounds to true, 
// performance diff to initializer list unknown, but likely negligible; should be measured if it matters
BoundingBoxType bbox (point1, point2); 
```

Reply:
@adayton1 thanks for confirming. That's what I would expect to happen.  What do you get if you do this?

`BoundingBox foo = {Point{}, Point{}}`

Maybe it is sufficient to note the difference in the comments??

Reply:
`BoundingBoxType foo = {Point{}, Point{}}` won't compile since both constructors are explicit.
`BoundingBoxType foo = BoundingBoxType{Point{}, Point{}}` will use initializer_list constructor.
`BoundingBoxType foo = BoundingBoxType(Point{}, Point{})` will use two point constructor.

Reply:
Thanks @adayton1 -- that looks good to me.

Presumably, this would be the same as your first line?:
```cpp
BoundingBoxType foo {Point{}, Point{}};
```

... but this would still work?
```cpp
BoundingBoxType foo (Point{}, Point{});
```

Reply:
Thanks for confirming @adayton1 That's exactly what I would expect. 

@kennyweiss they are different. Alan's first line (that doesn't compile) is copy-initialization, where as yours is direct-initialization. The first uses a copy/move ctor. The second uses a regular ctor.